projects/coordinator/src/main/java/org/batfish/coordinator/ApiKeyAuthenticationFilter.java

```diff
@@ -23,25 +19,19 @@ public void filter(ContainerRequestContext requestContext) {
         firstNonNull(
             requestContext.getHeaderString(CoordConsts.SVC_KEY_API_KEY),
             CoordConsts.DEFAULT_API_KEY);
-    try {
-      if (Strings.isNullOrEmpty(apiKey)) {
-        throw new IllegalArgumentException("apikey is missing or empty");
-      }
-
-      checkApiKeyValidity(apiKey);
-
-    } catch (Exception e) {
+    if (apiKey.isEmpty()) {
       requestContext.abortWith(
-          Response.status(Status.BAD_REQUEST)
-              .entity(e.getMessage())
+          Response.status(Status.UNAUTHORIZED)
+              .entity("ApiKey is empty")
+              .type(MediaType.APPLICATION_JSON)
+              .build());
+    } else if (!Main.getAuthorizer().isValidWorkApiKey(apiKey)) {
+      requestContext.abortWith(
+          Response.status(Status.FORBIDDEN)
```

Comment:
Actually, I am not sure whether unauthorized should be used here, since this filter checks whether the apikey is valid in general case, not about a specific resource. (This should be checked in the service class)